plugins/cilium-cni/lib/deletion_queue.go

```diff
@@ -138,7 +141,18 @@ func (dc *DeletionFallbackClient) EndpointDelete(id string) error {
 // either by directly accessing the API or dropping in a queued-deletion file.
 func (dc *DeletionFallbackClient) EndpointDeleteMany(req *models.EndpointBatchDeleteRequest) error {
 	if dc.cli != nil {
-		return dc.cli.EndpointDeleteMany(req)
+		err := dc.cli.EndpointDeleteMany(req)
+		status, ok := err.(runtime.ClientResponseStatus)
+		if !ok || !status.IsCode(http.StatusServiceUnavailable) {
+			return err
+		}
+
+		// Fallback to queuing endpoint delete if cilium Endpoint service is unavailable.
+		// Once ready, cilium-agent will process all the pending deletes in this queue.
+		dc.logger.Warn(
+			"Received ServiceUnavailable response from server, queuing delete request to be processed later",
+			logfields.Request, req,
+		)
```

Comment:
I'm not 100% sure it will work. 
Based on:
https://github.com/cilium/cilium/blob/a5952a8867b470cbd7a689977e6c32399ee36b71/plugins/cilium-cni/lib/deletion_queue.go#L44
it seems to me that `lockfile` below will be always nil in this case. 
So if we get connection working, but request returns 503, lockfile will be empty and we will never queue deletion?

Reply:
Ah yes, thats right. Seems like I missed re-acquiring the lock before exiting the `if`. I am working on adding tests for the changes, will fix this behavior.